### ISSUE TITLE
SNOW-735315: 'SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])"))'

### ISSUE THREAD

**Issue by carrerapcl** [snowflakedb/snowflake-connector-python#1421]
**1. What version of Python are you using?**

```
Python 3.9.16 (main, Jan 11 2023, 10:02:19) 
[Clang 14.0.6 ]
```

**2. What operating system and processor architecture are you using?**

`macOS-13.1-arm64-arm-64bit`

**3. What are the component versions in the environment (`pip freeze`)?**

```
agate==1.6.3
asn1crypto==1.5.1
attrs==22.2.0
Babel==2.11.0
betterproto==1.2.5
certifi @ file:///private/var/folders/nz/j6p8yfhx1mv_0grj5xl4650h0000gp/T/abs_5d968ni_yn/croot/certifi_1671487774636/work/certifi
cffi==1.15.1
charset-normalizer==2.1.1
click==8.1.3
colorama==0.4.5
cryptography==36.0.2
filelock==3.9.0
future==0.18.3
grpclib==0.4.3
h2==4.1.0
hologram==0.0.15
hpack==4.0.0
hyperframe==6.0.1
idna==3.4
importlib-metadata==6.0.0
isodate==0.6.1
jaraco.classes==3.2.3
Jinja2==3.1.2
jsonschema==3.2.0
keyring==23.13.1
leather==0.3.4
Logbook==1.5.3
MarkupSafe==2.0.1
mashumaro==3.3.1
minimal-snowplow-tracker==0.0.2
more-itertools==9.0.0
msgpack==1.0.4
multidict==6.0.4
networkx==2.8.8
oscrypto==1.3.0
packaging==21.3
parsedatetime==2.4
pathspec==0.10.3
pycparser==2.21
pycryptodomex==3.17
PyJWT==2.6.0
pyOpenSSL==22.0.0
pyparsing==3.0.9
pyrsistent==0.19.3
python-dateutil==2.8.2
python-slugify==8.0.0
pytimeparse==1.1.8
pytz==2022.7.1
PyYAML==6.0
requests==2.28.2
six==1.16.0
snowflake-connector-python @ https://github.com/snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
sqlparse==0.4.3
stringcase==1.2.0
text-unidecode==1.3
typing_extensions==4.4.0
urllib3==1.26.14
Werkzeug==2.1.2
zipp==3.12.0
```

Currently I downloaded `snowflake-connector-python` via `pip install --trusted-host github.com https://github.com/snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip`, as one of my attempts to find the issue or at least get a different error message. Also tried with `snowflake-connector-python==2.7.12` and `snowflake-connector-python==3.0.0`. It si worth noticing that `pip install github.com https://github.com/snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip`, without `--trusted-host`, also returned me a SSL error: 

```
Collecting https://github.com/snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)'))': /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)'))': /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
  WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)'))': /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
  WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)'))': /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
  WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)'))': /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip
ERROR: Could not install packages due to an OSError: HTTPSConnectionPool(host='github.com', port=443): Max retries exceeded with url: /snowflakedb/snowflake-connector-python/archive/refs/heads/master.zip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)')))
```

**4. What did you do?**

This error arised from an investigation on a 250001 error with the `dbt-snowflake` package. After several other checks, I narrowed down the problem to the `snowflake-connector-python` package and wrote a very simple script to test it out:

```
import snowflake.connector
import logging

logging.basicConfig(level=logging.DEBUG)

conn = snowflake.connector.connect(
    user='JOHN.DOE@COMPANY.CO',
    account='rt99999.eu-west-1',
    authenticator='externalbrowser',
    warehouse='DATA_ANALYSTS',
    database='JOHND',
    role='JOHND',
    schema='MARTS',
    insecure_mode=True,
    ssl_ca='/etc/ssl/cert.pem'
)
```

I tried to run it with several versions of the package (see last paragraph in previous question). Also, I did this in several different virtual environments, reinstalling packages from scratch, and also changing python versions and python installer (brew+virtualenvwrapper, and in the last try I did it with miniconda).

I can connect to Snowflake on the website without any problem, as well as with Tableau desktop, and with DBeaver.

I downloaded the certificate in app.snowflake.com and appended it to `/etc/ssl/cert.pem`, trying to see if that could solve the SSL issue, but nothing changed.

**5. What did you expect to see?**

Simply the connector establishing the connection successfully.

**6. Can you set logging to DEBUG and collect the logs?**

   ```
INFO:snowflake.connector.connection:Snowflake Connector for Python Version: 3.0.0, Python Version: 3.9.16, Platform: macOS-13.1-arm64-arm-64bit
DEBUG:snowflake.connector.connection:connect
DEBUG:snowflake.connector.connection:__config
INFO:snowflake.connector.connection:This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
INFO:snowflake.connector.connection:THIS CONNECTION IS IN INSECURE MODE. IT MEANS THE CERTIFICATE WILL BE VALIDATED BUT THE CERTIFICATE REVOCATION STATUS WILL NOT BE CHECKED.
INFO:snowflake.connector.connection:Setting use_openssl_only mode to False
DEBUG:snowflake.connector.converter:use_numpy: False
DEBUG:snowflake.connector.connection:REST API object was created: rt99999.eu-west-1.snowflakecomputing.com:443
DEBUG:keyring.backend:Loading KWallet
DEBUG:keyring.backend:Loading SecretService
DEBUG:keyring.backend:Loading Windows
DEBUG:keyring.backend:Loading chainer
DEBUG:keyring.backend:Loading libsecret
DEBUG:keyring.backend:Loading macOS
DEBUG:snowflake.connector.auth._auth:authenticate
DEBUG:snowflake.connector.auth._auth:assertion content: ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==
DEBUG:snowflake.connector.auth._auth:account=rt99999, user=JOHN.DOE@COMPANY.CO, database=JOHND, schema=MARTS, warehouse=DATA_ANALYSTS, role=JOHND, request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9
DEBUG:snowflake.connector.auth._auth:body['data']: {'CLIENT_APP_ID': 'PythonConnector', 'CLIENT_APP_VERSION': '3.0.0', 'SVN_REVISION': None, 'ACCOUNT_NAME': 'rt99999', 'LOGIN_NAME': 'JOHN.DOE@COMPANY.CO', 'CLIENT_ENVIRONMENT': {'APPLICATION': 'PythonConnector', 'OS': 'Darwin', 'OS_VERSION': 'macOS-13.1-arm64-arm-64bit', 'PYTHON_VERSION': '3.9.16', 'PYTHON_RUNTIME': 'CPython', 'PYTHON_COMPILER': 'Clang 14.0.6 ', 'OCSP_MODE': 'INSECURE', 'TRACING': 10, 'LOGIN_TIMEOUT': 120, 'NETWORK_TIMEOUT': None}, 'AUTHENTICATOR': 'ID_TOKEN', 'TOKEN': 'ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==', 'SESSION_PARAMETERS': {'CLIENT_PREFETCH_THREADS': 4, 'CLIENT_STORE_TEMPORARY_CREDENTIAL': True}}
DEBUG:snowflake.connector.auth._auth:Timeout set to 120
DEBUG:snowflake.connector.vendored.urllib3.util.retry:Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
DEBUG:snowflake.connector.vendored.urllib3.util.retry:Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
DEBUG:snowflake.connector.network:Session status for SessionPool 'rt99999.eu-west-1.snowflakecomputing.com', SessionPool 1/1 active sessions
DEBUG:snowflake.connector.network:remaining request timeout: 120, retry cnt: 1
DEBUG:snowflake.connector.network:Request guid: 3454c865-176c-44e2-b2de-50cda244dbc8
DEBUG:snowflake.connector.network:socket timeout: 60
DEBUG:snowflake.connector.vendored.urllib3.connectionpool:Starting new HTTPS connection (1): rt99999.eu-west-1.snowflakecomputing.com:443
DEBUG:snowflake.connector.vendored.urllib3.util.retry:Incremented Retry for (url='/session/v1/login-request?request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=3454c865-176c-44e2-b2de-50cda244dbc8'): Retry(total=0, connect=None, read=None, redirect=None, status=None)
WARNING:snowflake.connector.vendored.urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])"))': /session/v1/login-request?request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=3454c865-176c-44e2-b2de-50cda244dbc8
DEBUG:snowflake.connector.vendored.urllib3.connectionpool:Starting new HTTPS connection (2): rt99999.eu-west-1.snowflakecomputing.com:443
DEBUG:snowflake.connector.network:Hit non-retryable SSL error, HTTPSConnectionPool(host='rt99999.eu-west-1.snowflakecomputing.com', port=443): Max retries exceeded with url: /session/v1/login-request?request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=3454c865-176c-44e2-b2de-50cda244dbc8 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])")))
ERROR:snowflake.connector.network:Failed to get the response. Hanging? method: post, url: https://rt99999.eu-west-1.snowflakecomputing.com:443/session/v1/login-request?request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=3454c865-176c-44e2-b2de-50cda244dbc8, headers:{'Content-Type': 'application/json', 'accept': 'application/snowflake', 'User-Agent': 'PythonConnector/3.0.0 (macOS-13.1-arm64-arm-64bit) CPython/3.9.16', 'Content-Encoding': 'gzip'}, data: {"data": {"CLIENT_APP_ID": "PythonConnector", "CLIENT_APP_VERSION": "3.0.0", "SVN_REVISION": null, "ACCOUNT_NAME": "rt99999", "LOGIN_NAME": "JOHN.DOE@COMPANY.CO", "CLIENT_ENVIRONMENT": {"APPLICATION": "PythonConnector", "OS": "Darwin", "OS_VERSION": "macOS-13.1-arm64-arm-64bit", "PYTHON_VERSION": "3.9.16", "PYTHON_RUNTIME": "CPython", "PYTHON_COMPILER": "Clang 14.0.6 ", "OCSP_MODE": "INSECURE", "TRACING": 10, "LOGIN_TIMEOUT": 120, "NETWORK_TIMEOUT": null}, "AUTHENTICATOR": "ID_TOKEN", "TOKEN": "ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==", "SESSION_PARAMETERS": {"CLIENT_PREFETCH_THREADS": 4, "CLIENT_STORE_TEMPORARY_CREDENTIAL": true}}}
DEBUG:snowflake.connector.network:Session status for SessionPool 'rt99999.eu-west-1.snowflakecomputing.com', SessionPool 0/1 active sessions
DEBUG:snowflake.connector.connection:Operational Error raised at authenticationfor authenticator: AuthByIdToken
DEBUG:snowflake.connector.auth.by_plugin:Default timeout handler invoked for authenticator
DEBUG:snowflake.connector.auth.by_plugin:Hit connection timeout, attempt number 0. Will retry in a bit...
DEBUG:snowflake.connector.auth.by_plugin:Sleeping for 1 seconds
DEBUG:snowflake.connector.auth._auth:authenticate
DEBUG:snowflake.connector.auth._auth:assertion content: ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==
DEBUG:snowflake.connector.auth._auth:account=rt99999, user=JOHN.DOE@COMPANY.CO, database=JOHND, schema=MARTS, warehouse=DATA_ANALYSTS, role=JOHND, request_id=548029de-63db-4753-8684-309aa776753c
DEBUG:snowflake.connector.auth._auth:body['data']: {'CLIENT_APP_ID': 'PythonConnector', 'CLIENT_APP_VERSION': '3.0.0', 'SVN_REVISION': None, 'ACCOUNT_NAME': 'rt99999', 'LOGIN_NAME': 'JOHN.DOE@COMPANY.CO', 'CLIENT_ENVIRONMENT': {'APPLICATION': 'PythonConnector', 'OS': 'Darwin', 'OS_VERSION': 'macOS-13.1-arm64-arm-64bit', 'PYTHON_VERSION': '3.9.16', 'PYTHON_RUNTIME': 'CPython', 'PYTHON_COMPILER': 'Clang 14.0.6 ', 'OCSP_MODE': 'INSECURE', 'TRACING': 10, 'LOGIN_TIMEOUT': 120, 'NETWORK_TIMEOUT': None}, 'AUTHENTICATOR': 'ID_TOKEN', 'TOKEN': 'ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==', 'SESSION_PARAMETERS': {'CLIENT_PREFETCH_THREADS': 4, 'CLIENT_STORE_TEMPORARY_CREDENTIAL': True}}
DEBUG:snowflake.connector.auth._auth:Timeout set to 120
DEBUG:snowflake.connector.network:Session status for SessionPool 'rt99999.eu-west-1.snowflakecomputing.com', SessionPool 1/1 active sessions
DEBUG:snowflake.connector.network:remaining request timeout: 120, retry cnt: 1
DEBUG:snowflake.connector.network:Request guid: 33f1cb04-6383-41f1-bb23-1758fe931941
DEBUG:snowflake.connector.network:socket timeout: 60
DEBUG:snowflake.connector.vendored.urllib3.connectionpool:Starting new HTTPS connection (3): rt99999.eu-west-1.snowflakecomputing.com:443
DEBUG:snowflake.connector.vendored.urllib3.util.retry:Incremented Retry for (url='/session/v1/login-request?request_id=548029de-63db-4753-8684-309aa776753c&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=33f1cb04-6383-41f1-bb23-1758fe931941'): Retry(total=0, connect=None, read=None, redirect=None, status=None)
WARNING:snowflake.connector.vendored.urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])"))': /session/v1/login-request?request_id=548029de-63db-4753-8684-309aa776753c&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=33f1cb04-6383-41f1-bb23-1758fe931941
DEBUG:snowflake.connector.vendored.urllib3.connectionpool:Starting new HTTPS connection (4): rt99999.eu-west-1.snowflakecomputing.com:443
DEBUG:snowflake.connector.network:Hit non-retryable SSL error, HTTPSConnectionPool(host='rt99999.eu-west-1.snowflakecomputing.com', port=443): Max retries exceeded with url: /session/v1/login-request?request_id=548029de-63db-4753-8684-309aa776753c&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=33f1cb04-6383-41f1-bb23-1758fe931941 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])")))
ERROR:snowflake.connector.network:Failed to get the response. Hanging? method: post, url: https://rt99999.eu-west-1.snowflakecomputing.com:443/session/v1/login-request?request_id=548029de-63db-4753-8684-309aa776753c&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=33f1cb04-6383-41f1-bb23-1758fe931941, headers:{'Content-Type': 'application/json', 'accept': 'application/snowflake', 'User-Agent': 'PythonConnector/3.0.0 (macOS-13.1-arm64-arm-64bit) CPython/3.9.16', 'Content-Encoding': 'gzip'}, data: {"data": {"CLIENT_APP_ID": "PythonConnector", "CLIENT_APP_VERSION": "3.0.0", "SVN_REVISION": null, "ACCOUNT_NAME": "rt99999", "LOGIN_NAME": "JOHN.DOE@COMPANY.CO", "CLIENT_ENVIRONMENT": {"APPLICATION": "PythonConnector", "OS": "Darwin", "OS_VERSION": "macOS-13.1-arm64-arm-64bit", "PYTHON_VERSION": "3.9.16", "PYTHON_RUNTIME": "CPython", "PYTHON_COMPILER": "Clang 14.0.6 ", "OCSP_MODE": "INSECURE", "TRACING": 10, "LOGIN_TIMEOUT": 120, "NETWORK_TIMEOUT": null}, "AUTHENTICATOR": "ID_TOKEN", "TOKEN": "ver:1-hint:233461546513926-ETMsDgAAAYXA0ysiABRBRVMvQ0JDL1BLQ1M1UGFkZGluZwEAABAAEK/QWVdsN4zawi/bj/l5e64AAACQOCDtrgSO20YIvqI1MXviKBItqICpCADZrqpY4WFp679ZnLn+J5Ep//SOPURrUwJYpysXhbiu23kJDAEaj/PN1v4J6o+OMWsIbHxMDyBlSr/RbIJDLlpH9rFFK8QvD4vb/V8iprcVSaIG3Srnp96y4YcmBkjz0xOet4XFZsZ4Z49pEtOyn4m64CZqIYz9Wui/ABQJxwPTfWz0YuK5y8QBISzDbQP5/g==", "SESSION_PARAMETERS": {"CLIENT_PREFETCH_THREADS": 4, "CLIENT_STORE_TEMPORARY_CREDENTIAL": true}}}
DEBUG:snowflake.connector.network:Session status for SessionPool 'rt99999.eu-west-1.snowflakecomputing.com', SessionPool 0/1 active sessions
DEBUG:snowflake.connector.connection:Continuing authenticator specific timeout handling
DEBUG:snowflake.connector.auth.by_plugin:Default timeout handler invoked for authenticator
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 1105, in _authenticate
    auth.authenticate(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/auth/_auth.py", line 250, in authenticate
    ret = self._rest._post_request(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/network.py", line 706, in _post_request
    ret = self.fetch(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/network.py", line 804, in fetch
    ret = self._request_exec_wrapper(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/network.py", line 927, in _request_exec_wrapper
    raise e
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/network.py", line 858, in _request_exec_wrapper
    self._handle_unknown_error(method, full_url, headers, data, conn)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/network.py", line 987, in _handle_unknown_error
    Error.errorhandler_wrapper(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/errors.py", line 275, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/errors.py", line 333, in hand_to_other_handler
    connection.errorhandler(connection, cursor, error_class, error_value)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/errors.py", line 209, in default_errorhandler
    raise error_class(
snowflake.connector.errors.OperationalError: 250003: 250003: Failed to get the response. Hanging? method: post, url: https://rt99999.eu-west-1.snowflakecomputing.com:443/session/v1/login-request?request_id=f7345d1c-6976-4db7-8cfc-16e821c4b8a9&databaseName=JOHND&schemaName=MARTS&warehouse=DATA_ANALYSTS&roleName=JOHND&request_guid=3454c865-176c-44e2-b2de-50cda244dbc8

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/JOHN.DOE/Desktop/snowflake-connector/python-snowflake_connector.py", line 6, in <module>
    conn = snowflake.connector.connect(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/__init__.py", line 51, in Connect
    return SnowflakeConnection(**kwargs)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 302, in __init__
    self.connect(**kwargs)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 566, in connect
    self.__open_connection()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 828, in __open_connection
    self.authenticate_with_retry(self.auth_class)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 1084, in authenticate_with_retry
    self._authenticate(auth_instance)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 1150, in _authenticate
    raise auth_op from e
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/connection.py", line 1127, in _authenticate
    auth_instance.handle_timeout(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dbt/lib/python3.9/site-packages/snowflake/connector/auth/by_plugin.py", line 214, in handle_timeout
    raise OperationalError(
snowflake.connector.errors.OperationalError: 250001: 250001: Could not connect to Snowflake backend after 0 attempt(s).Aborting
   ```

Thank you for your help.

**Comment by garyzccisme** [comment]
Same issue with python connector:
```
SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])"))
```

I'm able to connect via website as well, only the python connector fails.

**Comment by sfc-gh-yixie** [comment]
@sfc-gh-mkeller I remember you worked on authentication not long ago. Is this related?

**Comment by sfc-gh-achandrasekaran** [comment]
Hello @carrerapcl - is this still happening? @sfc-gh-aling FYI

**Comment by cbolyard** [comment]
This could be related to SSL proxy config, I just had the same issue and found the below:

https://stackoverflow.com/questions/59151390/ssl-authentication-error-for-snow-client-when-trying-to-execute-from-command-lin

**Comment by pedrojrv** [comment]
I got the same stacktrace and same error with ssl certificate verify failed.

For me it was nothing related to SSL. When you copy your account from snowflake it introduces a dot (e.g., FDA3FS.AFDSA), I had to insert a dash instead (e.g., FDA3FS-AFDSA). Afterwards, we use private links in our org so we literally had to add privatelink so at the end, this is what worked for me (e.g., FDA3FS-AFDSA.privatelink)
